dashboard/app/models/script.rb

```diff
@@ -1331,13 +1331,49 @@ def self.rake
   # 1. Lesson names, which we get from the script DSL, and is passed in as lessons_i18n here
   # 2. Script Metadata (title, descs, etc.) which is in metadata_i18n
   # 3. Lesson descriptions, which arrive as JSON in metadata_i18n[:stage_descriptions]
-  def self.merge_and_write_i18n(lessons_i18n, unit_name = '', metadata_i18n = {})
+  def self.merge_and_write_i18n(lessons_i18n, unit_name = '', metadata_i18n = {}, log_event_type: 'write_other')
     units_yml = File.expand_path("#{Rails.root}/config/locales/scripts.en.yml")
+    old_size = `wc -l #{units_yml.dump}`.to_i
     i18n = File.exist?(units_yml) ? YAML.load_file(units_yml) : {}
 
     updated_i18n = update_i18n(i18n, lessons_i18n, unit_name, metadata_i18n)
-    unless i18n == updated_i18n
+    if i18n == updated_i18n
+      FirehoseClient.instance.put_record(
+        :analysis,
+        {
+          study: 'scripts_en_yml',
+          event: "skipped_#{log_event_type}",
+          data_string: unit_name,
+          user_id: current_user&.id,
+          data_json: {
+            old_size: old_size,
+            lessons_i18n: lessons_i18n,
+            metadata_i18n: metadata_i18n,
+          }.to_json
+        }
+      )
+    else
       File.write(units_yml, "# Autogenerated scripts locale file.\n" + updated_i18n.to_yaml(line_width: -1))
+
+      if log_event_type
```

Comment:
do we need this `if` for the above call, too?

Reply:
oops - good catch! now that I am providing a default value for `log_event_type`, I don't think I need it in either place